### ISSUE TITLE
Enable IMDSv2 requests for Visual Studio Code Remote - SSH

### ISSUE THREAD

**Issue by washingtoneg** [microsoft/vscode-remote-release#10134]
## Similar requests

This duplicates https://github.com/microsoft/vscode-remote-release/issues/6863.
 

## Feature description 

Currently, the SSH extension will send requests to AWS instance meta data service's v1 endpoint (IMDSv1).

Here is a blog from AWS post on why this is necessary / more secure to use _IMDSv2_:
* [Get the full benefits of IMDSv2 and disable IMDSv1 across your AWS infrastructure](https://aws.amazon.com/blogs/security/get-the-full-benefits-of-imdsv2-and-disable-imdsv1-across-your-aws-infrastructure/)

### Code changes requested
 
The following code is querying the metadat/v1 enpoint:

```
# .vscode-server/cli/servers/Stable-ea1445cc7016315d0f5728f8e8b12a45dc0a7286/server/out/server-main.js

if [ $? -eq 0 ]
then
	wget --connect-timeout=7 --tries=1 --dns-timeout=7 -q  -O - http://169.254.169.254/metadata/v1/id
else
	curl --version > /dev/null
	if [ $? -eq 0 ]
	then
		curl --connect-timeout 7 -s  http://169.254.169.254/metadata/v1/id
	fi
fi
exit 0
```

Can you enable IMDSv2 style queries for AWS target servers? This should work:

```
# .vscode-server/cli/servers/Stable-<new-version>/server/out/server-main.js

TOKEN=`curl -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 21600"`
if [ $? -eq 0 ]
then
	wget --header="X-aws-ec2-metadata-token: $TOKEN" \
	    --connect-timeout=7 --tries=1 --dns-timeout=7 -q  \
	    -O - http://169.254.169.254/latest/meta-data/instance-id
else
	curl --version > /dev/null
	if [ $? -eq 0 ]
	then
		curl --header="X-aws-ec2-metadata-token: $TOKEN" \
		    --connect-timeout 7 -s  http://169.254.169.254/latest/meta-data/instance-id
	fi
fi
exit 0
```

**Comment by washingtoneg** [comment]
Hello @roblourens,

Would you be able to provide an update on this issue? 

Thanks!

**Comment by joshspicer** [comment]
Taking a read of the extension code, it looks like we query both the v1 and v2 routes (specifically, For we associate AWS with the IMDS url  `http://169.254.169.254/latest/meta-data/instance-id` and for other clouds the v1 (http://169.254.169.254/metadata/v1/id). These requests are sent in parallel.

Is this causing any particular issues or raising any compliance flags where you need us to fix it.  I don't believe this should cause any difference functionally to you, to curious why you're interested in it :)

**Comment by primerano** [comment]
I thought this was a well written issue even though it is a duplicate of https://github.com/microsoft/vscode-remote-release/issues/6863 (both of which are closed)

IMDSv1 is insecure and should not be used anymore.   More and more organizations are blocking IMDSv1 outright and this may break vscode-remote. I honestly haven't looked at the code to see what will happen when the IMDSv1 request fails but I really think this issue should be reopened and fixed.

**Comment by joshspicer** [comment]
@primerano The extension is only querying these endpoints to identify possible clouds the remote VMs exist in. The query is not being used in any security-sensitive manner.  

The Remote - SSH extension will continue to work even if an organization blocks the IMDSv1 request.  With this information in mind, could you explain why you feel strongly this is worth changing?  I posed this question in December and I still don't quite understand (and am curious why!)

**Comment by primerano** [comment]
@joshspicer I thought vscode might fail if I disabled IMDSv1 but it still worked fine (as you mentioned ) so I guess there is no urgency to fix the code.   

That said, the code will be somewhat useless as new AMIs block it.  😉

Thanks for the quick response!
